microphone/drv_audio_pdm.c

```diff
@@ -101,8 +104,18 @@ static void drv_audio_pdm_event_handler(nrfx_pdm_evt_t const * const p_evt)
 			{
 				pdm_buf[l].released = true;
 				data_source_info.audio_source_info.audio_buffer = &pdm_buf[l].mic_buf;
+				if (l == 0 && timestamp_buffer[l] == 0) {
+					// First buffer, set timestamp to recording start timestamp
+					timestamp_buffer[l] = recording_start_timestamp;
+				} else {
+					// Subsequent buffers
+					// t_rec_i = t_release_i - (PDM_BUF_SIZE / effective_sampling_rate) * 1000
+					uint32_t base_sample_rate = 20000; // 20kHz
+					uint32_t effective_sampling_rate = drv_audio_get_mode() == 0 ? base_sample_rate/2 : base_sample_rate;
```

Comment:
I'd also like to know if this is actually a good approach for the timestamps. What is the difference with respect to the time stamps of the other sensors? Or with respect to assuming no drift from the first timestamp?